python/cudf/cudf/core/series.py

```diff
@@ -1419,7 +1419,7 @@ def as_mask(self):
         -------
         device array
         """
-        return cudautils.compact_mask_bytes(self.to_gpu_array())
+        return cudautils.compact_mask_bytes(self.data.mem)
```

Comment:
If self has nulls here you end up with undefined behavior, should likely fill nulls with `False` before this call.

Reply:
Moved it back to `self.to_gpu_array()` which leverages column for it's `fillna` function and should be fine as far as I can tell.

Reply:
Don't do that. By default the `fillna` value is `None` which just removes nulls from the array and returns a new dense buffer smaller than the original buffer and will cause strange undesired behavior.

In hindsight, this isn't the kernel that performs masking and just compresses a boolean array to a bit array, so we should assert there's no nulls here as well instead of filling with `False`.